pkg/gpu/context.go

```diff
@@ -34,8 +33,11 @@ type systemContext struct {
 	// deviceSmVersions maps each device index to its SM (Compute architecture) version
 	deviceSmVersions map[int]int
 
-	// fileData maps each executable file path to its Fatbin file data
-	fileData map[string]*fileData
+	// execData maps each executable file path to its Fatbin file data
+	execData map[string]*cuda.ExecutableData
+
+	// excDataLastUsed keeps track of the last time each executable data was used, for cleanup purposes
+	execDataLastUsed map[string]time.Time
```

Comment:
I think creating a wrapper struct that embeds the cuda.ExecutableData would be a more "cleaner" solution, instead of manipulating 2 different maps that coupled in logic